### ISSUE TITLE
[BUG] Task "tribe29.checkmk.agent : Create host on server." hangs up

### ISSUE THREAD

**Issue by RafaelKr** [tribe29/ansible-collection-tribe29.checkmk#159]
**Describe the bug**

I just tried to use this role to install the checkmk agent on my server an register it at my checkmk site.

The checkmk agent is installed successfully, but it hangs up at task "Create host on server."

My vars are set similar to this:
```
vars:
  checkmk_agent_version: "2.1.0p12"
  checkmk_agent_edition: cre
  checkmk_agent_protocol: https
  checkmk_agent_server: checkmk.example.com
  checkmk_agent_site: monitoring
  checkmk_agent_user: automation
  checkmk_agent_secret: '<MY_SECRET>'
  checkmk_agent_add_host: 'true'
  checkmk_agent_discover: 'true'
  checkmk_agent_folder: 'My Folder'
```

This is the output I get when running `ansible-playbook -vvv my_playbook.yaml`
```
TASK [tribe29.checkmk.agent : Create host on server.] ********************************************************************************************************************************
task path: /home/my_user/.ansible/collections/ansible_collections/tribe29/checkmk/roles/agent/tasks/main.yml:35
<localhost> ESTABLISH LOCAL CONNECTION FOR USER: my_user
<localhost> EXEC /bin/sh -c 'echo ~my_user && sleep 0'
<localhost> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/my_user/.ansible/tmp `"&& mkdir "` echo /home/my_user/.ansible/tmp/ansible-tmp-1663934876.4857767-36863-111637804248786 `" && echo ansible-tmp-1663934876.4857767-36863-111637804248786="` echo /home/my_user/.ansible/tmp/ansible-tmp-1663934876.4857767-36863-111637804248786 `" ) && sleep 0'
Using module file /home/my_user/.ansible/collections/ansible_collections/tribe29/checkmk/plugins/modules/host.py
<localhost> PUT /home/my_user/.ansible/tmp/ansible-local-3676448y416i9/tmpjrcqeokj TO /home/my_user/.ansible/tmp/ansible-tmp-1663934876.4857767-36863-111637804248786/AnsiballZ_host.py
<localhost> EXEC /bin/sh -c 'chmod u+x /home/my_user/.ansible/tmp/ansible-tmp-1663934876.4857767-36863-111637804248786/ /home/my_user/.ansible/tmp/ansible-tmp-1663934876.4857767-36863-111637804248786/AnsiballZ_host.py && sleep 0'
<localhost> EXEC /bin/sh -c 'sudo -H -S -p "[sudo via ansible, key=<HIDDEN>] password:" -u root /bin/sh -c '"'"'echo BECOME-SUCCESS-<HIDDEN> ; /usr/bin/python3 /home/my_user/.ansible/tmp/ansible-tmp-1663934876.4857767-36863-111637804248786/AnsiballZ_host.py'"'"' && sleep 0'
```

**Component Name**

`host`

**Ansible Version**

```console
$ ansible --version
ansible [core 2.13.4]
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/my_user/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3/dist-packages/ansible
  ansible collection location = /home/my_user/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/bin/ansible
  python version = 3.10.6 (main, Aug 10 2022, 11:40:04) [GCC 11.3.0]
  jinja version = 3.0.3
  libyaml = True

```

**Checkmk Version**

```console
e.g. 2.1.0p12 (CRE)
```

**Collection Version**

```console
$ ansible-galaxy collection list
tribe29.checkmk   0.10.0 
```
**Environment**

Local machine: Linux Mint 21 Cinnamon
Details: https://termbin.com/uvdqt

checkmk server and client: Debian 11

**To Reproduce**
2. Run '....'
3. Execute '....'

**Expected behavior**

**Actual behavior**

**Screenshots**

**Additional context**

**Comment by robin-tribe29** [comment]
The task to register the host runs on the host, that got the agent installed, or on your Ansible controller?
Can you confirm, that there is no firewall preventing communication? Because it looks like there is a dropping firewall in between.

**Comment by RafaelKr** [comment]
Hmmm that's a good question. The files which should be run seem to be created on my local machine.
There should be no firewall blocking it. From my machine I can connect to the checkmk server, its dashboard and also the agent host. (443 and 22 both are open).
Can I maybe somehow add a code line to debug where it hangs up?

`checkmk_agent_server` is the server where checkmk is running on, not the server where the agent is installed, correct?

**Comment by RafaelKr** [comment]
Also the previous step `tribe29.checkmk.agent : Debian Derivates: Install Checkmk CRE Agent.` is running correctly.

**Comment by RafaelKr** [comment]
This is my current full playbook.yaml

- hosts: default
  remote_user: ansible
  become: yes

  vars:
    checkmk_agent_version: "2.1.0p12"
    checkmk_agent_edition: cre
    checkmk_agent_protocol: https
    checkmk_agent_server: checkmk.example.com
    checkmk_agent_site: monitoring
    checkmk_agent_user: automation
    checkmk_agent_secret: '<MY_SECRET>'
    checkmk_agent_add_host: 'true'
    checkmk_agent_discover: 'true'
    checkmk_agent_folder: 'My Folder'

  vars_prompt:
    - name: "ansible_sudo_pass"
      prompt: "[sudo] password for {{ target_user }}"
      private: yes

  roles:
    - tribe29.checkmk.agent
    - setup-monitorin

Maybe its related to

```
- hosts: default
  remote_user: ansible
  become: yes
```

**Comment by RafaelKr** [comment]
I got something, when I go to `/home/my_user.ansible/collections/ansible_collections/tribe29/checkmk/roles/agent/tasks/main.yaml` on my local machine and set `become_user: my_user` on all tasks where `delegate_to` is set it works!

Although it doesn't find my folder. Maybe because of the space? It's like `checkmk_agent_folder: 'My Folder'`

**Comment by robin-tribe29** [comment]
I think it is relevant, where you run the registration task. But by default the role will run from the host, where the agent was just installed. So you want to make sure, that that host can talk to the Checkmk server.

**Comment by RafaelKr** [comment]
I now got a working solution, its fixed when explicitly adding `become: false` to these both tasks (the ones using `delegate_to: localhost`):
https://github.com/tribe29/ansible-collection-tribe29.checkmk/blob/96b0a246970939ba2174b8a9ef8279809f8b6272/roles/agent/tasks/main.yml#L35
https://github.com/tribe29/ansible-collection-tribe29.checkmk/blob/96b0a246970939ba2174b8a9ef8279809f8b6272/roles/agent/tasks/main.yml#L103

It wasn't working because ansible tried to connect with SSH to my local machine with the user `ansible`. Because I did set `become: yes` and `remote_user: ansible` on "global" level of my own playbook.
With the explicit `become: false` on the mentioned tasks this global config on my side is ignored.
Maybe this should be added to all places where `delegate_to: localhost` is used.

**Comment by robin-tribe29** [comment]
This is generally fixed by #169.
It might be necessary to double-check one's configuration, but with the added `become: false` the role should work out of the box.

**Comment by RafaelKr** [comment]
Checked the changes from the PR locally, its working for my case. Thanks!
In my last comment I stated I've added `become: false` to `tribe29.checkmk.discovery`. In the PR this wasn't changed and indeed, it is not required.

BTW in https://github.com/tribe29/ansible-collection-tribe29.checkmk/pull/169#issuecomment-1273293133 you wrote it will be merged for 0.12.0 but the next version would be 0.11.0. Did you mean that?
